File: commons-client/src/main/java/pl/chilldev/commons/client/interceptor/BearerTokenAuthorizer.java
/*
 * This file is part of the ChillDev-Commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2017 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package pl.chilldev.commons.client.interceptor;

import feign.RequestInterceptor;
import feign.RequestTemplate;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpHeaders;

/**
 * Feign request interceptor that injects custom HTTP authorization with Bearer token.
 */
@AllArgsConstructor
public class BearerTokenAuthorizer implements RequestInterceptor
{
    /**
     * Bearer token prefix.
     */
    private static final String TYPE_BEARER = "Bearer ";

    /**
     * HTTP access token..
     */
    private String token;

    /**
     * {@inheritDoc}
     */
    @Override
    public void apply(RequestTemplate requestTemplate)
    {
        // set the same value as incoming request to execute requests on behalf of the user
        requestTemplate.header(HttpHeaders.AUTHORIZATION, BearerTokenAuthorizer.TYPE_BEARER + this.token);
    }
}

File: commons-client/src/main/kotlin/pl/wrzasq/commons/client/interceptor/BearerTokenAuthorizer.kt
/*
 * This file is part of the pl.wrzasq.commons.
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2017, 2019 - 2021 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package pl.wrzasq.commons.client.interceptor

import feign.RequestInterceptor
import feign.RequestTemplate

/**
 * Authorization HTTP header name.
 */
const val HEADER_NAME_AUTHORIZATION = "Authorization"

private const val TYPE_BEARER = "Bearer "

/**
 * Feign request interceptor that injects custom HTTP authorization with Bearer token.
 *
 * @property token HTTP access token.
 */
class BearerTokenAuthorizer(
    private val token: String
) : RequestInterceptor {
    override fun apply(requestTemplate: RequestTemplate) {
        // to execute requests on behalf of the user set the same value as incoming request
        requestTemplate.header(
            HEADER_NAME_AUTHORIZATION,
            TYPE_BEARER + token
        )
    }
}
